docs/docsite/rst/contributor/DJANGO_REQUIREMENTS.rst

```diff
@@ -0,0 +1,843 @@
+
+    # Session security
+    SESSION_COOKIE_SECURE = True
+    SESSION_COOKIE_HTTPONLY = True
+    SESSION_COOKIE_SAMESITE = 'Lax'
+    SESSION_COOKIE_AGE = 1800
+
+    # CSRF protection
+    CSRF_COOKIE_SECURE = True
+    CSRF_COOKIE_HTTPONLY = True
+    CSRF_TRUSTED_ORIGINS = []
+
+6.2 External Secret Management
+------------------------------
+
+**REQUIRED**: Implement external secret loading:
+
+.. code-block:: python
+
+    # Secret key from external file
+    SECRET_KEY_FILE = os.environ.get('SECRET_KEY_FILE', '/etc/awx/SECRET_KEY')
+    if os.path.exists(SECRET_KEY_FILE):
+        with open(SECRET_KEY_FILE, 'rb') as f:
+            SECRET_KEY = f.read().strip().decode()
+    else:
+        if not DEBUG:
+            raise ImproperlyConfigured("SECRET_KEY must be configured in production")
+
+6.3 Proxy and Network Security
+------------------------------
+
+**REQUIRED**: Configure reverse proxy security:
+
+.. code-block:: python
+
+    # Proxy configuration
+    REMOTE_HOST_HEADERS = ['REMOTE_ADDR', 'REMOTE_HOST']
+    PROXY_IP_ALLOWED_LIST = []
+    USE_X_FORWARDED_HOST = True
+    USE_X_FORWARDED_PORT = True
+    SECURE_PROXY_SSL_HEADER = ('HTTP_X_FORWARDED_PROTO', 'https')
+
+**Requirements**:
+
+- External secret file management
+- Secure cookie configuration
+- CSRF protection with trusted origins
+- Proxy header validation
+- Force HTTPS in production
+
+----
+
+7. Database Management
+======================
+
+7.1 Advanced Database Configuration
+-----------------------------------
+
+**REQUIRED**: Optimize database connections for production:
+
+.. code-block:: python
+
+    # Database configuration with connection tuning
+    DATABASES = {
+        'default': {
+            'ENGINE': 'django.db.backends.postgresql',
+            'NAME': os.environ.get('DATABASE_NAME', 'awx'),
+            'ATOMIC_REQUESTS': True,
+            'CONN_MAX_AGE': 0,
+            'OPTIONS': {
+                'keepalives': 1,
+                'keepalives_idle': 5,
+                'keepalives_interval': 5,
+                'keepalives_count': 5,
+            },
+            'TEST': {
+                'NAME': 'test_awx',
+            },
+        }
+    }
+
+7.2 Database Version Validation
+-------------------------------
+
+**REQUIRED**: Implement database compatibility checking:
+
+.. code-block:: python
+
+    # PostgreSQL version enforcement
+    def validate_database_version():
+        from django.db import connection
+        if (connection.pg_version // 10000) < 12:
+            raise ImproperlyConfigured(
+                "PostgreSQL version 12 or higher is required"
+            )
+
+7.3 Migration Management
+------------------------
+
+**REQUIRED**: Structured migration organization::
+
+    migrations/
+    ├── 0001_initial.py
+    ├── 0002_squashed_v300_release.py
+    ├── 0003_squashed_v300_v303_updates.py
+    └── _migration_utils.py
+
+**Requirements**:
+
+- Use ``ATOMIC_REQUESTS`` for transaction safety
```

Comment:
This is going to be used by default given our production database settings. A contributor won't need to know about this.